Doc/library/io.rst

```diff
@@ -106,6 +106,54 @@ stream by opening a file in binary mode with buffering disabled::
 The raw stream API is described in detail in the docs of :class:`RawIOBase`.
 
 
+.. _io-text-encoding:
+
+Text Encoding
+-------------
+
+The default encoding of :class:`TextIOWrapper` and :func:`open` is
+locale-specific (:func:`locale.getpreferredencoding(False) <locale.getpreferredencoding>`).
+
+But many developers forget to specify encoding when opening text files
+encoded in UTF-8 (e.g. JSON, TOML, Markdown, etc...) since most Unix
+platforms uses UTF-8 locale by default. It cause a bug because locale-specific
+encoding is not UTF-8 for most Windows users. For example::
+
+   # may not work on Windows when non-ASCII characters in the file.
+   with open("README.md") as f:
+       long_description = f.read()
+
+Additionally, Python may change the default text encoding to UTF-8 in the
+future, although there is no plan yet.
+
+So it is highly recommended to specify encoding explicitly when opening text
+files. If you want to use UTF-8, specify ``encoding="utf-8"``. If you need to
+use locale-specific encoding, ``encoding="locale"`` is supported since Python
+3.10.
```

Comment:
```suggestion
Accordingly, it is highly recommended that you specify the encoding
explicitly when opening text files. If you want to use UTF-8, pass 
``encoding="utf-8"``. To use the current locale encoding,
``encoding="locale"`` is supported in Python 3.10.
```

Fix more grammar issues as above, avoid repetition/poor phrasing and clarify the description of the `"locale"` argument